documentation/en/default-lotus-config.toml

```diff
@@ -307,23 +292,62 @@
   # env var: LOTUS_EVENTS_MAXFILTERHEIGHTRANGE
   #MaxFilterHeightRange = 2880
 
-  # DatabasePath is the full path to a sqlite database that will be used to index actor events to
-  # support the historic filter APIs. If the database does not exist it will be created. The directory containing
-  # the database must already exist and be writeable. If a relative path is provided here, sqlite treats it as
-  # relative to the CWD (current working directory).
+
+[ChainIndexer]
+  # DisableIndexer controls whether the chain indexer is active.
+  # The chain indexer is responsible for indexing tipsets, messages, and events from the chain state.
+  # It is a crucial component for optimizing Lotus RPC response times.
+  # 
+  # Default: false (indexer is enabled)
+  # 
+  # Setting this to true will disable the indexer, which may significantly impact RPC performance.
+  # It is strongly recommended to keep this set to false unless you have a specific reason to disable it
+  # and fully understand the implications.
   #
-  # type: string
-  # env var: LOTUS_EVENTS_DATABASEPATH
-  #DatabasePath = ""
+  # type: bool
+  # env var: LOTUS_CHAININDEXER_DISABLEINDEXER
+  #DisableIndexer = false
 
+  # GCRetentionEpochs specifies the number of epochs for which data is retained in the Indexer.
+  # The garbage collection (GC) process removes data older than this retention period.
+  # Setting this to 0 disables GC, preserving all historical data indefinitely.
+  # 
+  # Default: 0 (GC disabled)
+  #
+  # type: int64
+  # env var: LOTUS_CHAININDEXER_GCRETENTIONEPOCHS
+  #GCRetentionEpochs = 0
 
-[Index]
-  # EXPERIMENTAL FEATURE. USE WITH CAUTION
-  # EnableMsgIndex enables indexing of messages on chain.
+  # ReconcileEmptyIndex determines whether to reconcile the index with the chain state
+  # during startup when the index is empty.
+  # 
+  # When set to true:
+  # - On startup, if the index is empty, the indexer will index the available
+  # chain state on the node albeit within the MaxReconcileTipsets limit.
+  # 
+  # When set to false:
+  # - The indexer will not automatically re-index the chain state on startup if the index is empty.
+  # 
+  # Default: false
+  # 
+  # Note: The number of tipsets reconciled (i.e. indexed) during this process can be
+  # controlled using the MaxReconcileTipsets option.
   #
   # type: bool
-  # env var: LOTUS_INDEX_ENABLEMSGINDEX
-  #EnableMsgIndex = false
+  # env var: LOTUS_CHAININDEXER_RECONCILEEMPTYINDEX
+  #ReconcileEmptyIndex = false
+
+  # MaxReconcileTipsets limits the number of tipsets to reconcile with the chain during startup.
+  # It represents the maximum number of tipsets to index from the chain state that are absent in the index.
+  # 
+  # Default: 3 * epochsPerDay (approximately 3 days of chain history)
+  # 
+  # Note: Setting this value too low may result in incomplete indexing, while setting it too high
+  # may increase startup time.
```

Comment:
Outsider question who hasn't dug into the code.  Do we give a progress bar / logging on startup for this operation?

Reply:
no, but we should, it would avoid lots of complaints; in `applyMissingTipsets`, maybe info log every 1000?

Reply:
some fraction of 2880 might be nice, but it's a weird number .. `%720` might be a neat division but it might just seem odd to have these strange multiples flying past, so 1000, or 500, something like that

Reply:
We do now show/log %progress here at every 2880 epochs. 

Please see updated `reconcile.go`